### ISSUE TITLE
Constructor fil format

### ISSUE THREAD

**Issue by jimmypw** [shlainn/game-file-formats#3]
You may be interested to know that today I implemented an extractor for constructor .fil files in C with help (and thanks) from your information.

https://github.com/jimmypw/constructor-fil

**Comment by RyanfaeScotland** [comment]
Good work! Great to see this game still getting love.

**Comment by RyanfaeScotland** [comment]
I'm going to look into this again. Interested? @jimmypw , @shlainn ?

**Comment by jimmypw** [comment]
I could be tempted. What were you thinking of looking in to?

**Comment by RyanfaeScotland** [comment]
I'm convinced the fence glitch bug must be apparent in the save file somewhere, I want to root it out and make a simple program that corrects it.

You and shlainn looked to have a much better grasp of how the data is structured than I did (my approach was a touch more brute force), I'd appreciate a little schooling on the FIL file format if you've the time?

I hope to user what I learn to make an extractor for the data from the save files, something I should be able to do anyway but probably better and faster with your input.

**Comment by shlainn** [comment]
Gday gentlemen,

I might be interested as well. It's been a while since I looked at those files though. Do you need anything in particular?

**Comment by RyanfaeScotland** [comment]
Ah nice, the band getting back together!

Just got a couple broad, foundation questions at the moment really. You discuss the .FIL archive format used in Constructor, but where do you actually find this file? I'll keep looking even after asking this, just keen to get up and running as quickly as possible and it doesn't appear to be in the game's directory. Do I need to extract it from elsewhere first?

**EDIT** - Found it, there is a file called const.gog which is actually an ISO, just change the extension to .iso and you can mount it and get access to the .fil files.

A lot of the linked sites are down now but the WayBack Machine has backups of the freeserver one and I managed to grab FilEdit from it again.

**Comment by shlainn** [comment]
Gentlemen, I just built a level viewer for constructor levels. As soon as github pages finishes deploying you can view it at [https://shlainn.github.io/game-file-formats/level.html](https://shlainn.github.io/game-file-formats/level.html).

So far I did not look at the entity data, but that will be the next thing to check out.

**Comment by RyanfaeScotland** [comment]
Damn man, can't wait!

I'm working on a file splitter / combiner for the save files, want to be able to parse out each of the sections you've described and work on them separately. Gave GSplit a try but it was erroring out.

**Comment by RyanfaeScotland** [comment]
> Gentlemen, I just built a level viewer for constructor levels. As soon as github pages finishes deploying you can view it at https://shlainn.github.io/game-file-formats/level.html.
> 
> So far I did not look at the entity data, but that will be the next thing to check out.

That's it available, looks great, good work!

**EDIT** - Just had a chance to look at the source for it as well and pretty excited to see it in HTML5. See, that's why we had to hold off on this project so long, the tech just wasn't ready for it yet. :D

I can see this being really useful for rending saves out of game, making it easier to see what is where. To that end, I've got the save splitter working, splits the files out into separate files for each chunk. Still needs cleaned up but will get it online sometime soon.
